### PR TITLE
Cleaned up renewal ASPD calculations

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #2067 and #2224

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Adjusted the ASPD table to match Aegis values.
  * Angra Manyu now disregards base ASPD values (Pre-Renewal only).
Thanks to Hercules for the base.

### CODE REVIEW

File: src/map/status.c [hunk 1]
```diff
@@ -2324,39 +2324,53 @@ int status_base_amotion_pc(struct map_session_data* sd, struct status_data* stat
...
 	// Percentual delay reduction from stats
-	amotion -= amotion * (4*status->agi + status->dex)/1000;
-#endif
+	amotion -= amotion * (4 * status->agi + status->dex) / 1000;
+
 	// Raw delay adjustment from bAspd bonus
 	amotion += sd->bonus.aspd_add;
 
+	// Angra Manyu disregards aspd_base and similar
+	if (pc_checkequip2(sd, ITEMID_ANGRA_MANYU, EQI_ACC_L, EQI_MAX))
```
Comment (Lemongrass3110) [inline src/map/status.c]: Move this to the top? Makes all other calculations useless?

**Comment by Lilori** [comment]
The ASPD is so close to iRO! Great job !
But there is a bug where equip dual-wielding (dual-dagger), aspd is 190 with 99 AGI.
cc : @aleos89

**Comment by aleos89** [comment]
@Lilori: Which class were you testing this on? This was on a normal class that doesn't have GM privileges to bypass requirements, right?

**Comment by Lilori** [comment]
@aleos89 Tested with Assassin Class (Assassin and Assassin Cross) with player previleges, only if equiping 2 dagger.

Status : 1 STR, 99 AGI, 1 VIT, 1 INT, 1 DEX, 1 LUK.
1 Dagger = Normal
1 Dagger + 1 Sword = Normal
2 Swords = Normal
2 Daggers = 190 ASPD (1 AGI, i got 180 ASPD).

Maybe needs additional 2HDagger in job_db1.txt ? XD

File: src/map/pc.h [hunk 2]
```diff
@@ -775,14 +775,19 @@ enum weapon_type {
 	W_GRENADE,	//21
 	W_HUUMA,	//22
 	W_2HSTAFF,	//23
-	MAX_WEAPON_TYPE,
+#ifndef MAX_SINGLE_WEAPON_TYPE
```
Comment (Jeybla) [inline src/map/pc.h]: Why the ifndefs?

**Comment by ecdarreola** [comment]
Tried this branch it seems like dual-wielding (dual-dagger) for Gx reduced ASPD that much. Like value of 10. Forgot to screenshot it.

**Comment by ecdarreola** [comment]
Where can I get some details for Renewal ASPD Calculations? Would like to test this PR.

**Comment by aleos89** [comment]
You could use RagState to test unless you have access to kRO.

**Comment by Jeybla** [comment]
@mazvi if you want to speed up the process of this pr, you're always invited to do some tests and share your results here. You could for example test the related issues: #2067, #2224, #2495.
Thanks in advance!

**Comment by ecdarreola** [comment]
Requesting to update this PR.

**Pull request merged**